### PR TITLE
OCPBUGS-31446: Add TrustedBundles to OAS container

### PR DESCRIPTION
**Pull request by jparrill**
**Which issue(s) this PR fixes**:
Fixes #[OCPBUGS-31446](https://issues.redhat.com/browse/OCPBUGS-31446)

### CODE REVIEW

File: control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go [hunk 1]
```diff
@@ -3126,8 +3126,21 @@ func (r *HostedControlPlaneReconciler) reconcileOpenShiftAPIServer(ctx context.C
 		}
 	}
 
+	var (
```
Comment (stevekuznetsov) [inline control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go:3129]: Why do we need the `var` block at all, could we just pass `hcp.Spec.AdditionalTrustBundle, hcp.Spec.Configuration`  to `oapi.ReconcileDeployment()`?

File: control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go [hunk 2]
```diff
@@ -3778,6 +3791,11 @@ func checkCatalogImageOverides(images ...string) (bool, error) {
 
 func (r *HostedControlPlaneReconciler) reconcileImageRegistryOperator(ctx context.Context, hcp *hyperv1.HostedControlPlane, releaseImageProvider *imageprovider.ReleaseImageProvider, userReleaseImageProvider *imageprovider.ReleaseImageProvider, createOrUpdate upsert.CreateOrUpdateFN) error {
 	params := registryoperator.NewParams(hcp, userReleaseImageProvider.Version(), releaseImageProvider, userReleaseImageProvider, r.SetDefaultSecurityContext)
+	rootCA := manifests.RootCAConfigMap(hcp.Namespace)
```
Comment (stevekuznetsov) [inline control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go:3794]: am I missing how `rootCA` is used after `r.Client.Get()`?

File: control-plane-operator/controllers/hostedcontrolplane/oapi/deployment.go [hunk 3]
```diff
@@ -185,6 +186,38 @@ func ReconcileDeployment(deployment *appsv1.Deployment, auditWebhookRef *corev1.
 		deployment.Spec.Template.ObjectMeta.Annotations[serviceCAHashAnnotation] = ""
 	}
 
+	var (
+		volumeProjection []corev1.VolumeProjection
+		additionalCA     bool
```
Comment (stevekuznetsov) [inline control-plane-operator/controllers/hostedcontrolplane/oapi/deployment.go:191]: can we simplify this by removing this variable and using `len(volumeProjection) > 0`?

Comment (jparrill) [reply control-plane-operator/controllers/hostedcontrolplane/oapi/deployment.go:191]: my bad, this was from another approach I used before.

File: control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go (hunk 2 above)
Comment (jparrill) [reply control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go:3794]: My bad, this was from another approach I used before.

File: control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go (hunk 1 above)
Comment (jparrill) [reply control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go:3129]: yep, you're right.

File: control-plane-operator/controllers/hostedcontrolplane/oapi/deployment.go (hunk 3 above)
Comment (jparrill) [reply control-plane-operator/controllers/hostedcontrolplane/oapi/deployment.go:191]: Makes sense, fixing it.

**Review by stevekuznetsov — APPROVED** [review]
/lgtm

File: control-plane-operator/controllers/hostedcontrolplane/oapi/deployment.go [hunk 4]
```diff
@@ -185,6 +191,34 @@ func ReconcileDeployment(deployment *appsv1.Deployment, auditWebhookRef *corev1.
 		deployment.Spec.Template.ObjectMeta.Annotations[serviceCAHashAnnotation] = ""
 	}
 
+	var volumeProjection []corev1.VolumeProjection
```
Comment (csrwng) [inline control-plane-operator/controllers/hostedcontrolplane/oapi/deployment.go:194]: nit: use a name that describes what the volume is for (like additionalTrusBundleVolume)

File: control-plane-operator/controllers/hostedcontrolplane/oapi/deployment.go [hunk 5]
```diff
@@ -185,6 +191,34 @@ func ReconcileDeployment(deployment *appsv1.Deployment, auditWebhookRef *corev1.
...
+		volumeProjection = append(volumeProjection, corev1.VolumeProjection{
+			ConfigMap: &corev1.ConfigMapProjection{
+				LocalObjectReference: corev1.LocalObjectReference{Name: clusterConf.Image.AdditionalTrustedCA.Name},
+			},
+		})
+	}
+
+	if len(volumeProjection) > 0 {
+		projVol := util.BuildVolume(additionalTrustBundleProjectedVolume(), buildAdditionalTrustBundleProjectedVolume)
+		projVol.Projected.Sources = volumeProjection
```
Comment (csrwng) [inline control-plane-operator/controllers/hostedcontrolplane/oapi/deployment.go:217]: should be done in the build function

**Review by csrwng — COMMENTED** [review]
A couple of comments

File: control-plane-operator/controllers/hostedcontrolplane/oapi/deployment.go (hunk 4 above)
Comment (jparrill) [reply control-plane-operator/controllers/hostedcontrolplane/oapi/deployment.go]: fixed

File: control-plane-operator/controllers/hostedcontrolplane/oapi/deployment.go (hunk 5 above)
Comment (jparrill) [reply control-plane-operator/controllers/hostedcontrolplane/oapi/deployment.go]: fixed

**Pull request merged**